File: app/src/main/java/com/hoymm/root/tictactoe2/GameEngine/GameFooterFragment.java
package com.hoymm.root.tictactoe2.GameEngine;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.hoymm.root.tictactoe2.DisplayMenu.MainMenu.Settings.DifficultyEnum;
import com.hoymm.root.tictactoe2.MainActivity;
import com.hoymm.root.tictactoe2.R;

/**
 * Created by hoymm on 01.11.17.
 */

public class GameFooterFragment extends Fragment{
    static final String HARDNESS_KEY = "com.hoymm.root.tictactoe2.GameEngine.HARDNESS_KEY";
    private Button exitButton;
    private TextView hardnessDisplayTextView;
    private String gameDifficulty;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        loadDataThatHasBeenSentToAFragment();
        return inflater.inflate(R.layout.game_footer, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initXMLObjects();
        setProperHardnessLevelText();
        setExitButtonAction();
    }

    private void loadDataThatHasBeenSentToAFragment() {
        loadDifficultyData();
    }

    private void loadDifficultyData() {
        DifficultyEnum difficultyEnum = (DifficultyEnum) getArguments().getSerializable(HARDNESS_KEY);
        assert difficultyEnum != null;
        switch(difficultyEnum){
            case easy:
                gameDifficulty = getString(R.string.easy);
                break;
            default:
            case medium:
                gameDifficulty = getString(R.string.medium);
                break;
            case hard:
                gameDifficulty = getString(R.string.hard);
                break;
        }
    }

    private void initXMLObjects() {
        exitButton = getActivity().findViewById(R.id.exitGameButtonId);
        hardnessDisplayTextView = getActivity().findViewById(R.id.hardnessDisplayTextId);
    }

    private void setProperHardnessLevelText() {
        hardnessDisplayTextView.setText(gameDifficulty);
    }

    private void setExitButtonAction() {
        exitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getContext(), MainActivity.class));
                getActivity().finish();
            }
        });
    }
}

File: app/src/main/java/com/hoymm/root/tictactoe2/GameEngine/GameActivityFragments/GameFooterFragment.kt
package com.hoymm.root.tictactoe2.GameEngine.GameActivityFragments

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

import com.hoymm.root.tictactoe2.DisplayMenu.MainMenu.Settings.DifficultyEnum
import com.hoymm.root.tictactoe2.GameEngine.CommunicationInterfaces.GameFragsCommunication
import com.hoymm.root.tictactoe2.MainActivity
import com.hoymm.root.tictactoe2.R

/**
 * Created by hoymm on 01.11.17
 */

class GameFooterFragment : Fragment() {
    private lateinit var gameFragsCommunication: GameFragsCommunication

    private var exitButton: Button? = null
    private var hardnessDisplayTextView: TextView? = null
    private var gameDifficulty: String? = null
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        loadDataThatHasBeenSentToAFragment()
        return inflater!!.inflate(R.layout.game_footer, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initXMLObjects()
        setProperHardnessLevelText()
        setExitButtonAction()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        gameFragsCommunication = context as GameFragsCommunication
    }

    private fun loadDataThatHasBeenSentToAFragment() {
        loadDifficultyData()
    }

    private fun loadDifficultyData() {
        val difficultyEnum = arguments.getSerializable(HARDNESS_KEY) as DifficultyEnum
        gameDifficulty = when (difficultyEnum) {
            DifficultyEnum.easy -> getString(R.string.easy)
            DifficultyEnum.medium -> getString(R.string.medium)
            DifficultyEnum.hard -> getString(R.string.hard)
        }
    }

    private fun initXMLObjects() {
        exitButton = activity.findViewById(R.id.exitGameButtonId)
        hardnessDisplayTextView = activity.findViewById(R.id.hardnessDisplayTextId)
    }

    private fun setProperHardnessLevelText() {
        hardnessDisplayTextView!!.text = gameDifficulty
    }

    private fun setExitButtonAction() {
        exitButton!!.setOnClickListener {
            startActivity(Intent(context, MainActivity::class.java))
            activity.finish()
        }
    }

    companion object {
        internal val HARDNESS_KEY = "com.hoymm.root.tictactoe2.GameEngine.HARDNESS_KEY"
    }
}
